src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -526,6 +525,34 @@ await invokeTask.NoThrowAwaitable(); // may or may not throw due to cancellation
             });
     }
 
+    [Fact]
+    public async Task CanInvokeServerMethodWithParameterPassedAsObject()
+    {
+        string result1 = await this.clientRpc.InvokeWithParameterPassedAsObjectAsync<string>(nameof(Server.TestParameter), new { test = "test" });
+        Assert.Equal("object", result1);
+    }
+
+    [Fact]
+    public async Task CanInvokeServerMethodWithParameterPassedAsArray()
+    {
+        string result1 = await this.clientRpc.InvokeAsync<string>(nameof(Server.TestParameter), "test");
+        Assert.Equal("array", result1);
+    }
+
+    [Fact]
+    public async Task CanInvokeServerMethodWithNoParameterPassedAsObject()
+    {
+        string result1 = await this.clientRpc.InvokeWithParameterPassedAsObjectAsync<string>(nameof(Server.TestParameter));
+        Assert.Equal("object or array", result1);
```

Comment:
I'm not sure I buy this -- until our support for *receiving* by-name messages includes finding method overloads with matching parameters. Until then, I think the only thing we should be willing to invoke is the one method that accepts `JToken`. So by that reasoning, I would expect `result1 == "object"`